test/adaptive_load/fake_plugins/fake_step_controller/fake_step_controller_test.cc

```diff
@@ -0,0 +1,187 @@
+#include "api/adaptive_load/benchmark_result.pb.h"
+#include "api/client/options.pb.h"
+#include "envoy/registry/registry.h"
+#include "adaptive_load/plugin_loader.h"
+#include "test/adaptive_load/fake_plugins/fake_step_controller/fake_step_controller.h"
+
+#include "external/envoy/source/common/config/utility.h"
+
+#include "gmock/gmock.h"
+#include "gtest/gtest.h"
+
+namespace Nighthawk {
+namespace {
+
+using ::nighthawk::adaptive_load::BenchmarkResult;
+using ::nighthawk::adaptive_load::FakeStepControllerConfig;
+using ::nighthawk::adaptive_load::MetricEvaluation;
+using ::nighthawk::client::CommandLineOptions;
+using ::testing::HasSubstr;
+
+TEST(FakeStepControllerConfigFactory, CreateEmptyConfigProtoCreatesCorrectType) {
+  auto& config_factory =
+      Envoy::Config::Utility::getAndCheckFactoryByName<StepControllerConfigFactory>(
+          "nighthawk.fake-step-controller");
+  Envoy::ProtobufTypes::MessagePtr message = config_factory.createEmptyConfigProto();
+  FakeStepControllerConfig expected_config;
+  EXPECT_EQ(message->DebugString(), expected_config.DebugString());
+}
+
+TEST(FakeStepControllerConfigFactory, FactoryRegistersUnderCorrectName) {
+  FakeStepControllerConfig config;
+  Envoy::ProtobufWkt::Any config_any;
+  config_any.PackFrom(config);
+  CommandLineOptions options;
+  auto& config_factory =
+      Envoy::Config::Utility::getAndCheckFactoryByName<StepControllerConfigFactory>(
+          "nighthawk.fake-step-controller");
+  EXPECT_EQ(config_factory.name(), "nighthawk.fake-step-controller");
+}
+
+TEST(FakeStepControllerConfigFactory, CreateMetricsPluginCreatesCorrectPluginType) {
```

Comment:
What does this have to do with the metrics plugin?

Reply:
Good catch, this was a typo.